Makefile

```diff
@@ -0,0 +1,35 @@
+
+#
+# This script installs the firmware files in the place where the linux
+# firmware loading system expects them - `/lib/firmware`. It supports
+# DESTDIR installations for package manager software. You should just
+# have to `sudo make install` to put the files in the correct location.
+#
+
+VERSION ?= v1.5.1
```

Comment:
any chance this could pick the git version from he current branch and be merged into master ? This way we only need to write it once.

Reply:
@marc-hb any suggestions?

Reply:
The usual solution is something like:
```shell
if test -f sof/.tarball-version; then
  # git-less or git override
   read values from sof/.tarball-version
else
  git describe --something --something
fi
```
We _already_ have CMake code in SOF that does that and saves the output somewhere. I'm trying it now.

Reply:
> We already have CMake code in SOF that does that and saves the output somewhere. I'm trying it now.

This requires fix https://github.com/thesofproject/sof/pull/3131 Merge it, then generate the .tarball-version file like this:

```shell
make -C sof/build/ dist # generates .tarball-version
```
and then VERSION is the first line of .tarball-version, which can be read like this:

```make
TARBALL_VERSIONS := $(file < build_apl_gcc/.tarball-version)
VERSION := $(word 1,${TARBALL_VERSIONS})
```

Or when using a release tarball instead of git drop the `build_apl_gcc/` directory:

```make
TARBALL_VERSIONS := $(file < .tarball-version)
VERSION := $(word 1,${TARBALL_VERSIONS})
```

Note however release tarballs have been broken since the conversion of rimage to a submodule (cc: @dcpleung ), see commit message of https://github.com/thesofproject/sof/pull/3132

Reply:
https://github.com/thesofproject/sof/pull/3131 is merged, so this works now:
```shell
make -C sof/build_apl_gcc/ rebuild_cache # needed only if you just changed the git version or tag
make -C sof/build_apl_gcc/ dist # generates sof/build_apl_gcc/.tarball-version
```
```make
# Read generated .tarball-version in this new Makefile
TARBALL_VERSIONS := $(file < sof/build_apl_gcc/.tarball-version)
VERSION := $(word 1,${TARBALL_VERSIONS}) # first word = first line

echo:
        # $VERSION}
```